fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -583,85 +215,24 @@ public static SearchParameter getSearchParameter(Class<?> resourceType, Canonica
-                    filterRules = allFilterRules.get(SearchConstants.WILDCARD_FILTER);
-                }
-
-                // If we found a non-empty list of search parameters to filter on, then do the filtering.
-                if (filterRules != null && !filterRules.isEmpty()) {
-                    boolean includeAll = false;
-
-                    for (Entry<String, String> filterRule : filterRules.entrySet()) {
-                        if (SearchConstants.WILDCARD_FILTER.equals(filterRule.getValue())) {
-                            includeAll = true;
-                        } else if (result.getCode().getValue().equals(filterRule.getKey())) {
-                            String url = result.getUrl().getValue();
-
-                            if (filterRule.getValue().equals(uri.getValue()) || filterRule.getValue().equals(url)) {
-                                return result;
-                            } else {
-                                // Configuration has mapped this code to a different search parameter
-                                return null;
-                            }
-                        }
-                    }
-
-                    // If we got here, then we havn't found an explicit rule for this search parameter
-                    // so filter out the result if there's no applicable wildcard rule
-                    if (!includeAll) {
-                        result = null;
-                    }
-                }
-            }
+        if (uri == null || uri.getValue() == null) {
+            return null;
         }
-        return result;
-    }
-
-    /**
-     * @param spMaps
-     * @param resourceType
-     * @param uri
-     * @return the SearchParameter for type {@code resourceType} with url {@code uri} or null if it doesn't exist
-     */
-    private static SearchParameter getSearchParameterByUrlIfPresent(Map<String, ParametersMap> spMaps, String resourceType, Canonical uri) {
-        SearchParameter result = null;
-
-        if (spMaps != null && !spMaps.isEmpty()) {
-            ParametersMap parametersMap = spMaps.get(resourceType);
-            if (parametersMap != null && !parametersMap.isEmpty()) {
-                result = parametersMap.lookupByCanonical(uri.getValue());
-            }
+        String tenantId = FHIRRequestContext.get().getTenantId();
+        Map<String, ParametersMap> paramsByResourceType = ParametersUtil.getTenantSPs(tenantId);
 
-            if (result == null) {
-                parametersMap = spMaps.get(SearchConstants.RESOURCE_RESOURCE);
-                if (parametersMap != null && !parametersMap.isEmpty()) {
-                    result = parametersMap.lookupByCanonical(uri.getValue());
+        // First try the passed resourceType, then fall back to the Resource resourceType (for whole system params)
+        for (String type : new String[]{resourceType, SearchConstants.RESOURCE_RESOURCE}) {
```

Comment:
new String[]{resourceType, SearchConstants.RESOURCE_RESOURCE}
Maybe this should be a private static final unmodifiable list

Reply:
resourceType is a method arg and so we'd need to have a constant for each one...I think its fine to create inline